prov/hook/trace/src/hook_trace.c

```diff
@@ -150,6 +150,34 @@ trace_cq(struct hook_cq *cq, const char *func, int line,
 	}
 }
 
+static inline void
+trace_cq_err(struct hook_cq *cq, const char *func, int line,
+             int count, struct fi_cq_err_entry *entry,  uint64_t flags)
+{
+	if ((count > 0) &&
+	    fi_log_enabled(cq->domain->fabric->hprov, FI_LOG_TRACE, FI_LOG_CQ)) {
+		if (entry->flags & FI_RECV) {
+			fi_log(cq->domain->fabric->hprov, FI_LOG_TRACE, FI_LOG_CQ,
+			       func, line,
+			       "ctx %p flags 0x%lx, len %zu buf %p data %lu tag %lu "
+			       "olen %zu err_code %d prov_err [%s]\n",
+			       entry->op_context, entry->flags, entry->len, entry->buf,
+			       entry->data, entry->tag, entry->olen,  entry->err,
+			       fi_cq_strerror(cq->hcq, entry->prov_errno, entry->err_data,
+			                      NULL, 0));
```

Comment:
It's okay to pass NULL into fi_cq_strerror, but the result means that we're not thread safe.  We'd be better off having a local ```char buf[80]``` variable, passing that to fi_cq_strerror(), and printing the results.  The buffer length is abitrary, but 80 should be sufficient.